app/src/home.js

```diff
@@ -177,7 +177,23 @@ export default function Home() {
   const history = useHistory();
 
   const addTranscriptChunk = useCallback(({ Alternatives, IsPartial, StartTime }) => {
-    const text = Alternatives[0].Transcript;
+    let addSpeakerLabel = true;
```

Comment:
This is really a reduction. Wondering if we can change to something like this for readability
```js
const [text] = Alternatives[0].Items.reduce(
  ([prevText, prevAddSpeakerLabel], item) => {
    const isSpeakerChange = item.Type === 'speaker-change';
    const shouldAddSpeakerLabel = !isSpeakerChange && prevAddSpeakerLabel && 'Speaker' in item;
    const isPronunciation = item.Type === 'pronunciation';
    const isPunctuation = item.Type === 'punctuation';

    const speakerLabel = shouldAddSpeakerLabel
      ? `
Speaker ${(parseInt(item.Speaker) + 1).toString()}
`
      : '';

    const itemContent = isPronunciation || isPunctuation ? item.Content : '';

    const spaceAtEnd = isPronunciation ? ' ' : '';

    const text = `${prevText}${speakerLabel}${itemContent}${spaceAtEnd}`;

    const addSpeakerLabel = isSpeakerChange || (shouldAddSpeakerLabel ? false : prevAddSpeakerLabel);

    return [text, addSpeakerLabel];
  },
  ['', true],
);
```

Reply:
Yep this is much cleaner. Updated